Lib/selectors.py

```diff
@@ -357,7 +357,7 @@ def register(self, fileobj, events, data=None):
             poller_events |= self._EVENT_WRITE
         try:
             self._selector.register(key.fd, poller_events)
-        except Exception:
+        except:
```

Comment:
> This is very uncommon code. If it is needed we have much more general problem.

In general it is needed when you catch BaseExceptions. That's why you don't want to catch them usually at all.

For this particular code, I think @haypo is right. I looked through the `unregister` method implementations and it looks like they don't raise exceptions (or can have unexpected ones). So probably a bare `except/raise` should work here.